### ISSUE TITLE
Missing date 

### ISSUE THREAD

**Issue by Sennahoi** [dmfs/lib-recur#31]
We expect the following code to generate the 2017-08-07:

```
RecurrenceRule rule = new RecurrenceRule("FREQ=WEEKLY;INTERVAL=1;BYDAY=MO");
DateTime ruleStartDate = new DateTime(2017, 1 - 1, 13, 0,0,0);
DateTime iteratorStartDate = new DateTime(2017, 8 - 1, 7, 0,0,0);
DateTime iterationEndDate = new DateTime(2017, 8 - 1, 8, 23, 59,59);
rule.setUntil(iterationEndDate);

RecurrenceRuleIterator it = rule.iterator(ruleStartDate);
it.fastForward(iteratorStartDate);

while (it.hasNext()) {
    DateTime nextInstance = it.nextDateTime();
    System.out.println(nextInstance);
}
```

But unfortunately it doesn't.

If we change iteratorStartDate to 2017-07-31 we get the expected list of
2017-07-31
2017-08-07

This seem to happen when the day of month is something less than 8 in every month in 2017. But this is only a guess. Do you have an idea what happens here?

**Comment by dmfs** [comment]
Right, it looks like a bug in `fastForward()` of `FastWeeklyIterator`. I'll check that.

**Comment by dmfs** [comment]
fixed in 0.9.8
